src/sentry/backup/dependencies.py

```diff
@@ -94,23 +103,41 @@ class PrimaryKeyMap:
     keys are not supported!
     """
 
-    mapping: dict[str, dict[int, int]]
+    mapping: dict[str, dict[int, Tuple[int, ImportKind]]]
 
     def __init__(self):
         self.mapping = defaultdict(dict)
 
-    def get(self, model: str, old: int) -> int | None:
+    def get_pk(self, model: str, old: int) -> int | None:
         """Get the new, post-mapping primary key from an old primary key."""
 
         pk_map = self.mapping.get(model)
         if pk_map is None:
             return None
-        return pk_map.get(old)
 
-    def insert(self, model: str, old: int, new: int):
+        entry = pk_map.get(old)
+        if entry is None:
+            return None
+
+        return entry[0]
+
+    def get_kind(self, model: str, old: int) -> ImportKind | None:
+        """Is the mapped entry a newly inserted model, or an already existing one that has been merged in?"""
+
+        pk_map = self.mapping.get(model)
+        if pk_map is None:
+            return None
+
+        entry = pk_map.get(old)
+        if entry is None:
+            return None
+
+        return entry[1]
+
+    def insert(self, model: str, old: int, new: int, kind: ImportKind):
```

Comment:
Is it worth to have a default value for kind? I'm assuming that would be `ImportKind.Inserted`

Reply:
I considered this, but decided against it: I'm pretty wary of default parameters like this, because it allows future callers of the function to not read the signature carefully and assume that setting the parameter is rare and optional, which it is isn't. I think a bit of extra typing and explicitness is a fair cost to pay for eliminating the bugs caused by incorrect defaults in the future.